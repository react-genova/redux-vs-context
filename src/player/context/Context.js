import { createContext } from 'react';
import { createSelector } from 'reselect';

const PlayerContext = createContext({
    name: '',
    surname: '',
    onNameChanged: () => null,
    onSurnameChanged: () => null
});

const getPlayer = value => value;
export const getPlayerName = createSelector(
    getPlayer,
    player => player.name
);
export const getPlayerSurname = createSelector(
    getPlayer,
    player => player.surname
);
export const getPlayerNameChangedCallback = createSelector(
    getPlayer,
    player => player.onNameChanged
);
export const getPlayerSurnameChangedCallback = createSelector(
    getPlayer,
    player => player.onSurnameChanged
);

export default PlayerContext;
