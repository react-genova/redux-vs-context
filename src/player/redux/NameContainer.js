import { connect } from 'react-redux';
import FieldShow from '../components/FieldShow';
import { getPlayerName } from './player';

export const mapStateToProps = state => ({ value: getPlayerName(state), label: 'Name' });

const NameContainer = connect(mapStateToProps)(FieldShow);

export default NameContainer;
