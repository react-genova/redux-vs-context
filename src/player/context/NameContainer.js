import React, { memo } from 'react';
import FieldShow from '../components/FieldShow';
import Context from './Context';

const PureFieldShow = memo(FieldShow);

const NameContainer = () => (
    <Context.Consumer>
    {({ name }) => <PureFieldShow value={name} label="Name" />}
    </Context.Consumer>
);

export default NameContainer;
