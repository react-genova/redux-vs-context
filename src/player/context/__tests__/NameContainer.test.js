import React from 'react';
import { render } from 'react-testing-library';
import NameContainer from '../NameContainer';
import Context, { noop } from '../Context';

jest.mock('../../components/FieldShow', () => ({
    __esModule: true,
    default: ({ value, label }) => (
        <div>
            <div data-testid="name">{value}</div>
            <div data-testid="label">{label}</div>
        </div>
    )
}));

describe('NameContainer consumer container', () => {
    it('renders without crashing and passes values down to its child', () => {
        const { getByTestId, container } = render((
            <Context.Provider
                value={{
                    name: 'Paolo',
                    surname: 'Rossi',
                    onNameChanged: noop,
                    onSurnameChanged: noop,
                }}
            >
                <NameContainer />
            </Context.Provider>
        ));
        expect(container.firstChild).toMatchSnapshot();
        expect(getByTestId('name').textContent).toBe('Paolo');
        expect(getByTestId('label').textContent).toBe('Name');
    });
});
