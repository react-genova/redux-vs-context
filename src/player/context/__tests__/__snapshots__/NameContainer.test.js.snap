// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameContainer consumer container renders without crashing and passes values down to its child 1`] = `
<div>
  <div
    data-testid="name"
  >
    Paolo
  </div>
  <div
    data-testid="label"
  >
    Name
  </div>
</div>
`;
